### PR TITLE
Allow `emptyInputBehavior` to be set to min, max, or a number

### PR DESCRIPTION
**Pull request by Apostrophii**
This implements https://github.com/autoNumeric/autoNumeric/issues/476.

It allows for `emptyInputBehavior` to be set to `min`, `max`, a number, or a string representing a number, in addition to the previous allowed values of `focus`, `press`, `always`, `null`, and `zero`.

### CODE REVIEW

**Comment by AlexandreBonneau** [comment]
This is great @Apostrophii !
With unit tests and everything!! :100:

I'll check that out tomorrow, thank you for your contribution :D

File: src/AutoNumeric.js [hunk 1]
```diff
@@ -6512,9 +6554,20 @@ To solve that, you'd need to either set \`decimalPlacesRawValue\` to \`null\`, o
                             this._triggerEvent(AutoNumeric.events.maxRangeExceeded, this.domElement);
                         }
                     }
-                } else if (rawValueToFormat === '' && this.settings.emptyInputBehavior === AutoNumeric.options.emptyInputBehavior.zero) {
-                    this._setRawValue('0');
-                    value = this.constructor._roundValue('0', this.settings, 0);
+                } else if (rawValueToFormat === '') {
+                    if (this.settings.emptyInputBehavior === AutoNumeric.options.emptyInputBehavior.zero) {
```
Comment (AlexandreBonneau) [inline src/AutoNumeric.js]: Please use a `switch` there too :)

File: src/AutoNumeric.js [hunk 2]
```diff
@@ -7337,17 +7390,30 @@ To solve that, you'd need to either set \`decimalPlacesRawValue\` to \`null\`, o
                     case AutoNumeric.options.emptyInputBehavior.focus:
                         setValue = false;
                         break;
-                    //TODO What about the `AutoNumeric.options.emptyInputBehavior.press` value?
+                    case AutoNumeric.options.emptyInputBehavior.null:
```
Comment (AlexandreBonneau) [inline src/AutoNumeric.js]: After checking your changes here, it occurred to me that I was setting `setValue` for no apparent reason in that block!

You can scrap those lines are just use the following instead:
```js
if (currentValue === '') {
    switch (this.settings.emptyInputBehavior) {
        case AutoNumeric.options.emptyInputBehavior.focus:
        case AutoNumeric.options.emptyInputBehavior.null:
        case AutoNumeric.options.emptyInputBehavior.press:
            break;
        case AutoNumeric.options.emptyInputBehavior.always:
            this._setElementValue(this.settings.currencySymbol);
            break;
        case AutoNumeric.options.emptyInputBehavior.min:
            this.set(this.settings.minimumValue);
            break;
        case AutoNumeric.options.emptyInputBehavior.max:
            this.set(this.settings.maximumValue);
            break;
        case AutoNumeric.options.emptyInputBehavior.zero:
            this.set('0');
            break;
        // When emptyInputBehavior is a number or a string representing a number
        default:
            this.set(this.settings.emptyInputBehavior);
    }
} else if (setValue && currentValue === this.domElement.getAttribute('value')) {
    this.set(currentValue);
}
```

File: src/AutoNumericOptions.js [hunk 3]
```diff
@@ -282,6 +282,8 @@ Object.defineProperty(AutoNumeric, 'options', {
                 focus : 'focus',
```
Comment (AlexandreBonneau) [inline src/AutoNumericOptions.js]: Please add a reference to the new options `min` and `max` in the JSDoc of this `emptyInputBehavior` option

File: test/unit/autoNumeric.spec.js [hunk 4]
```diff
@@ -1379,15 +1379,30 @@ describe('autoNumeric options and `options.*` methods', () => {
...
             expect(aNInput.getFormatted()).toEqual('');
+            aNInput.options.reset().french().set('');
             aNInput.options.emptyInputBehavior(AutoNumeric.options.emptyInputBehavior.focus);
             expect(aNInput.getFormatted()).toEqual('');
+            aNInput.options.reset().french().set('');
             aNInput.options.emptyInputBehavior(AutoNumeric.options.emptyInputBehavior.always);
             expect(aNInput.getFormatted()).toEqual('\u202f€');
+            aNInput.options.reset().french().set('');
+            aNInput.options.emptyInputBehavior(AutoNumeric.options.emptyInputBehavior.min);
+            expect(aNInput.getFormatted()).toEqual('-9.999.999.999.999,99\u202f€');
```
Comment (AlexandreBonneau) [inline test/unit/autoNumeric.spec.js]: This is great :)

Could you please add a test for each `rawValue` starting from here in that test block?

For example for the line above:
```js
expect(aNInput.getNumericString()).toEqual('-9999999999999.99');
```

File: test/unit/autoNumeric.spec.js [hunk 5]
```diff
@@ -4503,6 +4574,30 @@ describe('Instantiated autoNumeric functions', () => {
             expect(aNInput.getNumericString()).toEqual('0');
         });
 
+        it('should return number from `emptyInputBehavior` setting as the default value', () => {
```
Comment (AlexandreBonneau) [inline test/unit/autoNumeric.spec.js]: Note to self: I need to ditch the DOM element creation in each test and use `beforeEach`/`afterEach`..

**Comment by AlexandreBonneau** [comment]
Do you think you'll be able to checkout the changes is the next few days @Apostrophii ?

I'd prefer not having to overwrite your changes and let you do that on your end.

**Comment by Apostrophii** [comment]
Yeah, I'll see if I can take care of this before the weekend. Sorry I haven't been more active, it's been a busy summer

**Comment by AlexandreBonneau** [comment]
Hey @Apostrophii, no problem and no pressure, I feel you ; the summer has been busy around here too :)

**Comment by AlexandreBonneau** [comment]
This is live in `v4.4.0`.

Thanks again for this PR @Apostrophii !

**Pull request merged**